LeaderboardBackend/Authorization/MiddlewareResultHandler.cs

```diff
@@ -15,9 +15,9 @@ public class MiddlewareResultHandler : IAuthorizationMiddlewareResultHandler
         PolicyAuthorizationResult policyAuthorizationResult
     )
     {
-        if (policyAuthorizationResult.Forbidden)
+        if (policyAuthorizationResult.AuthorizationFailure?.FailureReasons.Any(fr => fr.Handler is UserTypeAuthorizationHandler) ?? false)
         {
-            httpContext.Response.StatusCode = (int)HttpStatusCode.NotFound;
+            httpContext.Response.StatusCode = (int)HttpStatusCode.Unauthorized;
```

Comment:
I have no idea at this point. Everything about this auth system feels like a hack. The way I have it now seems to make sense. If it can't validate your JWT, it returns 401 because the server doesn't know who you are.